### PR TITLE
Wheelchairs

### PR DESCRIPTION
**Pull request by unid15**
Adds wheelchairs to the game (sprites from Baystation)

While they don't give as much mobility as hyperzine does, they allow players without legs (or otherwise unable to stand) to properly use computers and pick up items.
You need at least one hand to use a wheelchair. The more hands you have, the faster you go.
Holding items in hands decreases the wheelchair's speed. Holding stuff in both hands makes you drive as if you were one-handed, being one-handed while holding an item makes you very slow etc.

Wheelchairs can be found in medbay (which is provided two at roundstart) or ordered from cargo for 40$ each.

Characters with both their feet missing start with a wheelchair

whwelcah

Monkeys and xenos can use them as well (but since the wheelchair isn't fit for them, they use it 2x slower, and are unable to use it with both hands occupied)

other changes:
all vehicles now have a speed variable which can be changed to make them slower (or faster)
all vehicles now have a maximum health variable (previously only clowncart had one)

### CODE REVIEW

**Comment by PJB3005** [comment]
>all floor tiles have randomized pixel_x and pixel_y (previously only metal floor tiles had this 'feature')

Atomic PRs hur hur

otherwise :+1:

**Comment by Exxion** [comment]
>The more hands you have, the faster you go.

It sounds like you're suggesting that you could move at extreme speed if you somehow acquired more than two hands.

File: code/game/objects/structures/vehicles/wheelchair.dm [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	//Returns a number from 0 to 4 depending on usability of user's hands
+	//User with no hands gets 0
+	//User with one hand holding something gets 1
+	//User with one empty hand gets 2
+	//User with two hands both holding something gets 2
+	//User with one empty and one full hand gets 3
+	//User with two empty hands gets 4
+
+	//Wheelchair's speed depends on the resulting value
+	var/mob/living/carbon/human/H = user
```
Comment (ComicIronic) [inline code/game/objects/structures/vehicles/wheelchair.dm]: Why do you assume the user is human?
Indent the organ checks under an istype(H)

**Comment by NigglyWiggly** [comment]
>all vehicles now have a speed variable which can be changed to make them slower (or faster)

Time to sprite a Transam 20000 replacement for the janicart

File: code/game/objects/structures/vehicles/wheelchair.dm (hunk 1 above)
Comment (Unfit) [inline code/game/objects/structures/vehicles/wheelchair.dm]: Pretty sure non humans can't use vehicles

Comment (PJB3005) [inline code/game/objects/structures/vehicles/wheelchair.dm]: muh adminbus

**Comment by Unfit** [comment]
honk

**Comment by PJB3005** [comment]
that looks fun

**Comment by clusterfack** [comment]
If you want a define to be viable in all files put it in setup.dm

File: code/game/objects/structures/vehicles/wheelchair.dm (hunk 1 above)
Comment (ComicIronic) [inline code/game/objects/structures/vehicles/wheelchair.dm]: I believe monkeys can. Regardless, you shouldn't make a typecast without a typecheck, typically.

File: code/game/jobs/job_controller.dm [hunk 2]
```diff
@@ -440,6 +440,13 @@ var/global/datum/controller/occupations/job_master
 				var/obj/item/clothing/glasses/G = H.glasses
 				G.prescription = 1
 //		H.update_icons()
+		//Gives wheelchair to those missing both of their feet
+		var/datum/organ/external/left_leg = H.get_organ("l_foot")
+		var/datum/organ/external/right_leg = H.get_organ("r_foot")
+
+		if(!left_leg || left_leg.status & ORGAN_DESTROYED && !right_leg || right_leg.status & ORGAN_DESTROYED) //If the character is missing both of his feet
```
Comment (ComicIronic) [inline code/game/jobs/job_controller.dm]: You have to wrap each organ check in parentheses, because && is evaluated before || and also for readability.

**Comment by NigglyWiggly** [comment]
Well now I guess I need to make a crippled Xeno sprite replacement.

**Comment by bustygoku** [comment]
Are the sprites ported from somewhere?

**Comment by Ephx1** [comment]
@bustygoku
Straight from Baystation I believe

**Comment by Unfit** [comment]
yeah, they're from bay

**Comment by d3athrow** [comment]
Atomic PRs

**Pull request merged**